### PR TITLE
chore: upgrade dependencies

### LINKED ISSUE #78: Update dependencies [Q4]
howto: https://github.com/mozilla-services/autopush/issues/1170#issuecomment-378776031

I may attempt to finish my previous effort from last quarter to upgrade to hyper 0.12

### PR DESCRIPTION
**Pull request by pjenvey**
split in 2 commits for review

Closes #78

Also see https://github.com/mozilla-services/autopush-rs/compare/chore/78...chore/hyper0.12-take2 and its commit message for my 2nd attempt to upgrade hyper (because misery loves company)

It appears to work but there's some caveats w/ ClientRegistry's locking that I don't want to spend any more time on at the moment

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -586,7 +585,7 @@ fn save_and_notify_undelivered_messages(
                 } else if let Some(node_id) = user.node_id {
                     let result = AsyncClient::builder()
                         .timeout(Duration::from_secs(1))
-                        .build(&srv3.handle);
+                        .build();
```
Comment (bbangert) [inline src/client.rs]: The async client uses its own event loop then?

Comment (pjenvey) [reply src/client.rs]: w/ the tokio reform, going forward you don't really pass Handles around. Now there's kind of a global event loop. Taskish things generally get an executor to schedule themself from a globalish CurrentThread executor or DefaultExecutor, I'm not entirely sure which one in this case.

But basically it should be running on the current executor (which sets itself as one of those when it starts)

Comment (bbangert) [reply src/client.rs]: That's a lot of ish-ish! ;)

Comment (pjenvey) [reply src/client.rs]: very ish-ish hand waivey, I don't have a good grip on all this yet

**Pull request merged**